### PR TITLE
#339 Fix problem with nil keys

### PR DESCRIPTION
**Pull request by Zavierazo**
Fix problem with nil keys. Currently nil keys are writed as empty byte instead of nil

### CODE REVIEW

File: write.go [hunk 1]
```diff
@@ -621,7 +621,12 @@ func (wb *writeBuffer) writeRecord(attributes int8, baseTime time.Time, offset i
 	wb.writeVarInt(int64(milliseconds(timestampDelta)))
 	wb.writeVarInt(offsetDelta)
 
-	wb.writeVarBytes(msg.Key)
+	if msg.Key != nil {
+		wb.writeVarBytes(msg.Key)
```
Comment (achille-roussel) [inline write.go]: Should we handle this case in the `writeVarBytes` method instead? It could prevent future issues of the same kind.

Comment (Zavierazo) [reply write.go]: Good idea ;) I will do and test it ;)

Comment (Zavierazo) [reply write.go]: Change done and tested.
This also will fix the tombstone used for kafka with log compaction.

**Comment by achille-roussel** [comment]
Thanks for your contribution!

**Pull request merged**